### ISSUE TITLE
Batched example_inputs

### ISSUE THREAD

**Issue by grigorn** [VainF/Torch-Pruning#264]
For certain pruning schemes a big example input may provide better estimate of which weights should be pruned, but forward pass  
on big dataset may not be possible due to memory limits. Is there a functionality for example_inputs to be a dataloader or smth similar to do batched inference? Thanks

**Comment by VainF** [comment]
Hi @grigorn, one solution is to iterate the train_loader for several steps and accumulate the gradients for taylor/hessian pruning. For example:

https://github.com/VainF/Torch-Pruning/blob/8a9c81fde2d1533ae7f7afa17e6a4f9f683fe176/examples/transformers/prune_timm_vit.py#L154

**Comment by grigorn** [comment]
I was asking about pruning part as to get importances of each function, pruner does one forward pass on whole network. Is there equivalent of gradient accumulation for weight importance?
